youtube_dl/extractor/pornhub.py

```diff
@@ -169,15 +230,11 @@ def _real_extract(self, url):
         host = mobj.group('host') or 'pornhub.com'
         video_id = mobj.group('id')
 
-        if 'premium' in host:
-            if not self._downloader.params.get('cookiefile'):
-                raise ExtractorError(
-                    'PornHub Premium requires authentication.'
-                    ' You may want to use --cookies.',
-                    expected=True)
-
         self._set_cookie(host, 'age_verified', '1')
 
+        # Authenticate, if required
+        self._login_if_required(host)
```

Comment:
This should be in `_real_initialize`. Same for all other occurrences.

Reply:
Yeah, that would be ideal but, this can't be in `_real_initialize` because we need to know the URL you're accessing so we can load the credentials for pornhub.com or pornhubpremium.com.